CMakeLists.txt

```diff
@@ -84,11 +90,15 @@ set(SOURCE_FILES
 source_group(include FILES ${INCLUDE_FILES})
 source_group(source FILES ${SOURCE_FILES})
 
-add_library(enet STATIC
+add_library(enet
     ${INCLUDE_FILES}
     ${SOURCE_FILES}
 )
 
-if (MINGW)
+if (WIN32)
+    if (BUILD_SHARED_LIBS)
+        set_property(TARGET enet PROPERTY WINDOWS_EXPORT_ALL_SYMBOLS ON)
```

Comment:
I've never used `WINDOWS_EXPORT_ALL_SYMBOLS`, but it sounds bad.
I'd be concerned with polluting the namespace; I personally prefer explicit exports of the symbols that should be exported.

Reply:
I am aware of CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS but `enet` does not mark its API's with dllexport attribute.
We can add .def which is a pain to maintain or add dllexport attribute which I think is a good alternative option.

Reply:
Notice that this property only affects Win32 DLL files. A enet.dll file on windows with all symbols exported and compiled with `/MT` is about 700kb which I don't find big.